### PR TITLE
chore(site): Make tests faster

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
- Use @swc/jest instead of ts-jest
- Allow GH Actions to use more workers
- Upgrade Jest
- Scoped a few `ByRole` queries to make the queries faster

FYI: Compared with `main`, this branch runs tests 50% faster on GitHub Actions.

### CODE REVIEW

File: site/jest.config.js [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-//         If we don't limit it, then 100% CPU and high MEM usage is hit
-//         unexpectedly, leading to OOM kills.
-//
-// SEE thread: https://github.com/coder/coder/pull/483#discussion_r829636583
-const maxWorkers = 2
-
 module.exports = {
-  maxWorkers,
+  testTimeout: 10_000,
+  maxWorkers: 8,
```
Comment (kylecarbs) [inline site/jest.config.js:3]: Isn't this value going to be set in CI anyways? It'd be ideal to use all 64 cores in dogfood

**Review by kylecarbs — APPROVED** [review]

File: site/src/components/DeploymentBanner/DeploymentBannerView.tsx [hunk 2]
```diff
@@ -196,7 +196,6 @@ export const DeploymentBannerView: FC<DeploymentBannerViewProps> = ({
         <Tooltip title="A countdown until stats are fetched again. Click to refresh!">
           <Button
             className={`${styles.value} ${styles.refreshButton}`}
-            title="Refresh"
```
Comment (johnstcn) [inline site/src/components/DeploymentBanner/DeploymentBannerView.tsx:199]: Is this expected?

File: site/src/components/Loader/FullScreenLoader.tsx [hunk 3]
```diff
@@ -20,7 +20,7 @@ export const FullScreenLoader: FC = () => {
   const styles = useStyles()
 
   return (
-    <div className={styles.root}>
+    <div className={styles.root} data-testid="loader">
```
Comment (Kira-Pilot) [inline site/src/components/Loader/FullScreenLoader.tsx:23]: Is it still slow if we keep querying using `getByRole` but also use within?

**Review by Kira-Pilot — APPROVED** [review]

File: site/jest.config.js (hunk 1 above)
Comment (BrunoQuaresma) [reply site/jest.config.js:3]: Yes, but when it happens into 64 cores at the same time, the output is kinda crazy. I think we can remove it if it is too slow tho but with 8 workers, it is kinda fast.

Comment (kylecarbs) [reply site/jest.config.js:3]: Makes sense!

**Pull request merged**